x-pack/metricbeat/module/gcp/metrics/_meta/docs.asciidoc

```diff
@@ -30,6 +24,27 @@ Please see
 https://cloud.google.com/monitoring/api/ref_v3/rpc/google.monitoring.v3#aligner[Aggregation Aligner]
 for the full list of aligners.
 
+* *metric_types*: Required, a list of metric type strings, or a list of metric
+type prefixes. For example, `instance/cpu` is the prefix for metric type
+`instance/cpu/usage_time`, `instance/cpu/utilization` etc Each call of the
+`ListTimeSeries` API can return any number of time series from a single metric
+type. Metric type is to used for identifying a specific time series.
+
+* *service*: Required, the name of the service for related metrics. This should
+be a valid Google Cloud service name. Service names may be viewed from the
+corresponding page from https://cloud.google.com/monitoring/api/metrics[GCP Metrics list documentation].
+The `service` field is used to compute the GCP Metric prefix, unless
+`service_metric_prefix` is set.
+
+* *service_metric_prefix*: A string containing the full Metric prefix as
```

Comment:
In GCP context, this seems to be called "[API service name](https://cloud.google.com/apis/design/glossary#api_service_name)", I would suggest to use this terminology and call this option `service_name` or `api_service_name`.

It may sound redundant with `service`, but in that case `service` may be deprecated at some moment. If `service` is used for some other thing, maybe we can add a field for this some other thing, as `service_namespace` if it is used as a "namespace" for the metrics collected for this service. Or we can rename it to [`service_title`](https://cloud.google.com/apis/design/glossary#api_title).

Being it a "prefix" is an implementation detail, users should only care about the name of the service they want to monitor.

The `/` after the value looks also a bit artificial, service names are described without them in the GCP docs.

Reply:
The reasoning is correct, but I find it even more confusing, due to how is presented in the GCP documentation.

For example, all "Google Cloud" metrics have an explicit relation to the concept (i.e. pubsub => "pubsub.googleapis.com"); but still, there is nowhere written that the PubSub API service name is that one. I tried searching for it, but still the term "API service name" is nowhere to be seen.
That poses a problem when you reach GKE metric collection, where the prefix is "kubernetes.io".

The only place where the "kubernetes.io" string [appears](https://cloud.google.com/monitoring/api/metrics_kubernetes) is in this phrase:
> The "metric type" strings in this table must be prefixed with <code translate="no" dir="ltr">kubernetes.io/</code>. That prefix has been omitted from the entries in the table.

That's the reason I choose `service_metric_prefix`, I still think is somewhat confusing (what is a "metric prefix"?) but has direct reference in the GCP documentation. The disadvantage is that exposes an implementation detail of the GCP Monitoring APIs, but:
- all Google Cloud metrics do not require it due to the default implementation
- I expect we are going to provide dedicated metricsets for the most used GCP metrics, greatly reducing user exposure to this detail

Reply:
Ok, if this currently only needs to b used for gke, and they use the prefix terminology there, it sounds good to me.